prototype/src/gpu/dust.hpp

```diff
@@ -339,6 +365,41 @@ class Dust {
   real_t** _particle_y_addrs;
   real_t** _particle_y_swap_addrs;
   uint64_t* _rng_state;
+
+  void initialise(const init_t data, const size_t step,
+                  const size_t n_particles) {
+    _particles.clear();
+    _particles.reserve(n_particles);
+
+    cudaFree(_particle_y_addrs);
```

Comment:
This will error when this is called by the constructor, but should be ok when called by `reset()`. This stops the error state being checked. One thought would be to call `cudaDeviceReset()` here, but that would destroy the RNG state.
So I would instead explicitly set these three pointers to zero at the start of the constructor, which means that `cudaFree()` performs no operation, and then you can reintroduce the `cdpErrchk()` wrapper

Reply:
Empirically this works for me as it's default constructed as NULL - are you getting an error with your compiler?